pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -2172,6 +2172,9 @@ protected void handleNewTxn(CommandNewTxn command) {
                     }
                     ctx.writeAndFlush(Commands.newTxnResponse(requestId, txnID.getLeastSigBits(),
                             txnID.getMostSigBits()));
+                    if (getBrokerService().getInterceptor() != null) {
+                        getBrokerService().getInterceptor().beginTxn(command.getTcId(), txnID.toString());
+                    }
```

Comment:
It's better to move to `PulsarCommandSender` so that we can have a unified management of sent commands and interception

Reply:
Hi @codelipenghui , 
   I have added unit tests and refactored code as you had recommended. Can you please review again?